src/main/java/edu/ksu/canvas/attendance/form/CourseConfigurationValidator.java

```diff
@@ -14,28 +14,40 @@ public boolean supports(Class<?> clazz) {
 
     @Override
     public void validate(Object target, Errors errors) {
+
         CourseConfigurationForm courseConfigurationForm = (CourseConfigurationForm) target;
+
+        String assignmentPoints = courseConfigurationForm.getAssignmentPoints();
+        String presentPoints = courseConfigurationForm.getPresentPoints();
+        String tardyPoints = courseConfigurationForm.getTardyPoints();
+        String absentPoints = courseConfigurationForm.getAbsentPoints();
+        String excusedPoints = courseConfigurationForm.getExcusedPoints();
+
+
         if (courseConfigurationForm.getTotalClassMinutes() < courseConfigurationForm.getDefaultMinutesPerSession() && !errors.hasFieldErrors("totalClassMinutes")) {
             errors.rejectValue("defaultMinutesPerSession", "ExceedTotal.courseConfigurationForm.defaultMinutesPerSession");
         }
 
         if(courseConfigurationForm.getGradingOn()) {
             if (courseConfigurationForm.getAssignmentName() == null || courseConfigurationForm.getAssignmentName().length() <= 1 || courseConfigurationForm.getAssignmentName().trim().isEmpty()) {
                 errors.rejectValue("assignmentName", "Assignment Name is required.");
+                return;
             }
 
-            if (courseConfigurationForm.getAssignmentPoints() == null || courseConfigurationForm.getAssignmentPoints() <= 0) {
+            if (assignmentPoints == null || (Double.parseDouble(assignmentPoints) < 0)) {
                 errors.rejectValue("assignmentPoints", "Total Points is a required field and must be greater than 0.");
+                return;
             }
 
-            if (courseConfigurationForm.getPresentPoints() == null || courseConfigurationForm.getTardyPoints() == null || courseConfigurationForm.getExcusedPoints() == null
-                    || courseConfigurationForm.getAbsentPoints() == null) {
+            if (presentPoints == null || tardyPoints == null || absentPoints == null || excusedPoints == null) {
                 errors.rejectValue("presentPoints", "All status point fields are required.");
                 return;
             }
 
-            if ((courseConfigurationForm.getPresentPoints() > 100 || courseConfigurationForm.getPresentPoints() < 0) || (courseConfigurationForm.getTardyPoints() > 100 || courseConfigurationForm.getTardyPoints() < 0)
-                    || (courseConfigurationForm.getExcusedPoints() > 100 || courseConfigurationForm.getExcusedPoints() < 0) || (courseConfigurationForm.getAbsentPoints() > 100 || courseConfigurationForm.getAbsentPoints() < 0)) {
+            if ((Double.parseDouble(presentPoints) > 100) || (Double.parseDouble(presentPoints) < 0)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of conditional operators (7) used in the expression (maximum allowed 3). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1067)